examples/leasing/node_release_script.py

```diff
@@ -0,0 +1,241 @@
+#! /bin/python
+"""
+This script will free nodes from a project after
+a specific amount of time.
+Script reads a config file from /etc/leasing.cfg
+which defines which nodes are in the project, time threshold
+admin user name and password and status file.
+
+status file includes each nodes status. which node is in
+the project, which one is free and for how long the node
+has been in the project.
+"""
+
+from sets import Set
+import time
+import ConfigParser
+from os import remove
+from shutil import move
+
+from hil.client.client import Client, RequestsHTTPClient
+from hil.client.base import FailedAPICallException
+
+
+class HILClientFailure(Exception):
+    """Exception indicating that the HIL client failed"""
+
+
+def hil_client_connect(endpoint_ip, name, pw):
+    """Connect to the HIL server and return a HIL Client instance"""
+
+    hil_http_client = RequestsHTTPClient()
+    if not hil_http_client:
+        print('Unable to create HIL HTTP Client')
+        return None
+
+    hil_http_client.auth = (name, pw)
+
+    return Client(endpoint_ip, hil_http_client)
+
+
+def release_nodes(
+        statusfile, non_persistent_list,
+        threshold_time, hil_username, hil_password, hil_endpoint,
+        hil_client=None
+        ):
+
+    """Release nodes : Takes node list as arguments
+    and puts them back to free pool after given time"""
+
+    if not hil_client:
+        hil_client = hil_client_connect(
+                        hil_endpoint, hil_username, hil_password)
+
+    free_node_list = hil_client.node.list('free')
+    # Only these nodes should be updated in
+    # the file(either for project or for time)
+    nodes_to_update_infile = list(Set(
+        non_persistent_list)-Set(free_node_list))
+
+    for node in nodes_to_update_infile:
+        # get the information from status file for node.
+        node_info_from_file = get_project_and_time_for_node(statusfile, node)
```

Comment:
I think I suggested this when you made the initial request for feedback on your design: it would be better to just parse the file once up front, then do all of your calculations on some data structure (maybe something like `{'node_name': {'threshold': 4, 'project': 'my-project'}, ...}`?), then write it back out all at once.

Additionally, when I find myself coming up with variables that are this long, I tend to step back and ask if perhaps I'm doing things that conceptually shouldn't be coupled. If your function name is a full sentence, then you haven't really encapsulated much.

If you use the representation I suggest above, you might end up with something like:

```python
node_data = load_node_data(filename)
...

to_update = ...
for node in to_update:
    old_project = node['project']
    old_time = node['time']
    ...
```

Note that because `nodes_to_update_infile` is only used for the span of 2 statements, you probably don't need such a long name for clarity; the extra info is clear from the surrounding context.